op-node/rollup/engine/engine_controller.go

```diff
@@ -107,22 +110,31 @@ type EngineController struct {
 
 func NewEngineController(engine ExecEngine, log log.Logger, metrics ECMetrics,
 	rollupCfg *rollup.Config, syncCfg *sync.Config, emitter event.Emitter,
+) *EngineController {
+	return NewEngineControllerWithEventFields(engine, log, metrics, rollupCfg, syncCfg, emitter, nil, nil)
+}
```

Comment:
If we merge the two metrics structs, we can just introduce a `ctx` to the `NewEngineController` and generally it is always a good idea to include a context even you don't have one - `context.Background()` or `context.TODO()`, rather than `nil`.